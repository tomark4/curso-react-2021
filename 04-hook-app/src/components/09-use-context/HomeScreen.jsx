import { useContext } from "react"
import {UserContext} from './UserContextProvider'

const HomeScreen = () => {
    const {user} = useContext(UserContext);

    return (
        <div>
            <h1>Home</h1>
            <pre>
                {JSON.stringify(user)}
            </pre>
        </div>
    )
}

export default HomeScreen
