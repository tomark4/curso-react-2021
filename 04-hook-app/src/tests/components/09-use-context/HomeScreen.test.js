import { shallow, mount } from "enzyme";
import HomeScreen from '../../../components/09-use-context/HomeScreen'
import UserContextProvider from "../../../components/09-use-context/UserContextProvider";
describe('Pruebas en home screen component', () => {
  const user = {
    name: 'jose',
    email: 'jotta@jotta.com'
  };

  test('should de mostrarse el componente correctamente', () => {
    const wrapper = mount(
      <UserContextProvider value={{user}}>
        <HomeScreen />
      </UserContextProvider>
    )

    expect(wrapper).toMatchSnapshot()
  })


});
