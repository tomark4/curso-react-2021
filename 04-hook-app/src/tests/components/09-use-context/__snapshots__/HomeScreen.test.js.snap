// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en home screen component should de mostrarse el componente correctamente 1`] = `
<div>
  <h1>
    Home
  </h1>
  <pre />
</div>
`;
